docs/developing/package-managers/README.md

```diff
@@ -0,0 +1,48 @@
+# Package Managers
+
+It is useful to divide software into a few broad categories based on how it is used and best managed.
+
+## General Purpose Software
+
+The most common general purpose software are command line tools that are used across many projects and don't require specific versions. Some examples include [direnv](https://direnv.net/), [entr](http://eradman.com/entrproject/limits.html), and [jq](https://stedolan.github.io/jq/), along with alternative shells like [fish](https://fishshell.com) or desktop applications such as [Postico](https://eggerapps.at/postico/).
+
+In general, we recommend using the native package manager supplied by your operating system to install this kind of software. On a Mac, use [homebrew](https://brew.sh).
+
+:::note
+
+In situations where a project requires a specific version or build of what would otherwise be considered general purpose, that software should not be considered general purpose within the context of the project.
+
+We don't recommend using homebrew to install anything that a project will need a specific version of.
+
+:::
+
+## Language Runtimes
+
+Use [asdf](package-managers/asdf) to install and activate specific versions of language runtimes. You will need to also setup the corresponding [language plugin](https://asdf-vm.com/manage/plugins.html) for the language you want to manage with asdf.
+
+## Language-specific Dependency Managers
+
+Many programming languages have their own package managers to install and manage project dependencies. We recommend using the following dependency managers:
+
+- [npm](https://www.npmjs.com/) for JavaScript and TypeScript (Node)
+- [Poetry](https://python-poetry.org/) for Python
+- [RubyGems](https://rubygems.org/) for Ruby
+- `go mod` for Go
```

Comment:
I think you do a good job of explaining the distinction between a package manager and a dependency manager. Since this page is about package managers, it could be argued that we shouldn't have recommendations about dependency managers here as those are truly language-specific tools.

Reply:
As someone who doesn't understand the difference between npm, yarn, npx, pnpm, I would love a page like this.

Reply:
I don't know if we need to split up the difference into different pages of documentation. They're still managing dependencies and packaged software regardless of it's based on the chip architecture, operating system, or programming language. I think there's a good breakdown of what these differences are and what these tools are. I do see this page as being a catch-all for managing packages/dependencies. Once we need to differentiate between language specific and chip/OS specific managers, we can further flesh this out into separate pages. This page of documentation is under 50 lines so I think it's good enough for it to be as general as it is.